### ISSUE TITLE
[question] Define the number of steps per episode during training

### ISSUE THREAD

**Issue by AjarAilerons** [araffin/rl-baselines-zoo#63]
The class _TimeFeatureWrapper_ has a max_steps parameter controlling the number of steps per episode, is there a way of modifying this parameter when using the training script without modifying it explicitly in the _TimeFeatureWrapper_ class?

**Comment by araffin** [comment]
> is there a way of modifying this parameter when using the training script without modifying it explicitly in the TimeFeatureWrapper class?

For now, no, unless you have a custom environment. In that case, it will use the value from the  `TimeLimit` wrapper.

**Comment by AjarAilerons** [comment]
I see. In the environments that don't use the class TimeFeatureWrapper, the number of steps per episode is defined as the corresponding hyperparameter _n_steps_, right?

**Comment by araffin** [comment]
Not really, TimeFeatureWrapper is made for envs that use the TimeLimit wrapper and thus break the Markov assumption.
I am not sure to which n_steps you are refering, maybe ppo, then take a look at the doc for that.

**Comment by AjarAilerons** [comment]
According to the [documentation](https://stable-baselines.readthedocs.io/en/master/modules/ppo2.html#stable_baselines.ppo2.PPO2) _n_steps_ is "the number of steps to run for each environment per update". I'm not sure if by _update_, you meant episode. If not, I can't find where number of steps per episodes used in the _train.py_ script is defined.

**Comment by araffin** [comment]
update means gradient update, I recommend you reading more about ppo and policy gradient in general (we have some link in the doc, spinning up guide is a good start) to understand what it means exactly ;)

**Comment by AjarAilerons** [comment]
Will do! :) So the original question: where exactly is defined the number of steps per episode -aka episode length- that is used in the training script?

**Comment by araffin** [comment]
>where exactly is defined the number of steps per episode -aka episode length- that is used in the training script?

steps per episode is defined by the environment not the training, for instance for CarPole, it is defined here:
https://github.com/openai/gym/blob/master/gym/envs/__init__.py#L63

**Comment by AjarAilerons** [comment]
Solved, thank you!
